core/metacling/src/TCling.cxx

```diff
@@ -1779,8 +1779,14 @@ void TCling::RegisterModule(const char* modulename,
       // its symbols are not yet reachable from the process.
       // Recursive dlopen seems to work just fine.
       dyLibName = FindLibraryName(triggerFunc);
+
       if (dyLibName) {
-         // We were able to determine the library name.
+         // The triggerFunc may not be in a shared object but in an executable.
+         // In that case, we are done.
+         if (!cling::DynamicLibraryManager::isSharedLibrary(dyLibName))
+            return;
```

Comment:
I am still convinced that this is not the intended behavior: we should also register the dictionary for binaries, e.g. inject the fwd decls, declare the headers etc.

Reply:
I agree. That means more refactoring.